services/provider/server/server.go

```diff
@@ -188,3 +204,143 @@ func newClient() (client.Client, error) {
 
 	return client, nil
 }
+func (s *OCSProviderServer) getJSONConnectionDetails(consumerResource *ocsv1alpha1.StorageConsumer) ([]*pb.ExternalResource, error) {
+	var extR []*pb.ExternalResource
```

Comment:
you should be able to use the same context that is passed into the `GetStorageConfig` function

Reply:
updated